asserts/database_test.go

```diff
@@ -1576,6 +1576,80 @@ func (safs *signAddFindSuite) TestFindSequence(c *C) {
 
 }
 
+func (safs *signAddFindSuite) TestCheckConstraints(c *C) {
+	headers := map[string]interface{}{
+		"type":         "account",
+		"authority-id": "canonical",
+		"account-id":   "my-brand",
+		"display-name": "My Brand",
+		"validation":   "verified",
+		"timestamp":    time.Now().Format(time.RFC3339),
+	}
+	acct, err := safs.signingDB.Sign(asserts.AccountType, headers, nil, safs.signingKeyID)
+	c.Assert(err, IsNil)
+
+	err = safs.db.Add(acct)
+	c.Check(err, IsNil)
+
+	pubKey1 := testPrivKey1.PublicKey()
+	pubKey1Encoded, err := asserts.EncodePublicKey(pubKey1)
+	c.Assert(err, IsNil)
+
+	now := time.Now().UTC()
+	headers = map[string]interface{}{
+		"authority-id":        "canonical",
+		"format":              "1",
+		"account-id":          "my-brand",
+		"public-key-sha3-384": pubKey1.ID(),
+		"name":                "default",
+		"since":               now.Format(time.RFC3339),
+		"until":               now.AddDate(1, 0, 0).Format(time.RFC3339),
+		"constraints": []interface{}{
+			map[string]interface{}{
+				"headers": map[string]interface{}{
+					"type":  "model",
+					"model": "foo-.*",
+				},
+			},
+		},
+	}
+	accKey, err := safs.signingDB.Sign(asserts.AccountKeyType, headers, []byte(pubKey1Encoded), safs.signingKeyID)
+	c.Assert(err, IsNil)
+
+	err = safs.db.Add(accKey)
+	c.Check(err, IsNil)
+
+	headers = map[string]interface{}{
+		"type":         "model",
+		"authority-id": "my-brand",
+		"brand-id":     "my-brand",
+		"series":       "16",
+		"model":        "foo-200",
+		"classic":      "true",
+		"timestamp":    now.Format(time.RFC3339),
+	}
+	mfoo, err := asserts.AssembleAndSignInTest(asserts.ModelType, headers, nil, testPrivKey1)
+	c.Assert(err, IsNil)
+
+	headers = map[string]interface{}{
+		"type":         "model",
+		"authority-id": "my-brand",
+		"brand-id":     "my-brand",
+		"series":       "16",
+		"model":        "goo-200",
+		"classic":      "true",
+		"timestamp":    now.Format(time.RFC3339),
+	}
+	mnotfoo, err := asserts.AssembleAndSignInTest(asserts.ModelType, headers, nil, testPrivKey1)
+	c.Assert(err, IsNil)
+
+	err = safs.db.Add(mfoo)
```

Comment:
(super nitpick) maybe move this safs.db.Add() right after when mfoo is created? so that it's clear one is the happy and the other the unhappy case ? but feel free to ignore of course

Reply:
done